connector/oauth.go

```diff
@@ -133,35 +133,44 @@ type CreateAccessTokenJson struct {
 	GrantType    string `json:"grant_type"`
 	ClientID     string `json:"client_id"`
 	ClientSecret string `json:"client_secret"`
+	Code         string `json:"code"`
 }
 
 func createAccessTokenHandler(c *FakeConnector, capturer *RequestCapturer) http.HandlerFunc {
 	return func(rw http.ResponseWriter, req *http.Request) {
 		id, secret, ok := req.BasicAuth()
-		var grantTypeStr string
-		if !ok {
-			if hasFormValues(req) {
+		var grantTypeStr, code string
+		if hasFormValues(req) {
+			if req.FormValue("client_id") != "" && req.FormValue("client_secret") != "" {
 				id = req.FormValue("client_id")
 				secret = req.FormValue("client_secret")
-				grantTypeStr = req.FormValue("grant_type")
-			} else if req.Body != nil {
-				// Assume JSON
-				decoder := json.NewDecoder(req.Body)
-				data := &CreateAccessTokenJson{}
-				if err := decoder.Decode(data); err != nil {
-					fmt.Println("Failed to decode JSON: ", err)
-					connector.ToOAuthError(err).WriteResponse(rw, jsonProducer)
-					return
-				}
+			}
+			if req.FormValue("code") != "" {
+				code = req.FormValue("code")
+			}
+			grantTypeStr = req.FormValue("grant_type")
+		} else if req.Body != nil {
+			// Assume JSON
```

Comment:
minor: without knowing how must of this works - could we also switch based on the content-type header and return an error if we encounter an unexpected value? probably not important at all if those two are the only content-types we'll accept

Reply:
I think we're doing this because there isn't a real standard for which content_type to give when sending json. Some send `text/json`, others `application/json`.

Since this is nothing more than some fake code for the providers to test with, I'm not sure it would be that important to validate the content-type. Swagger does it anyway I believe.